### PR TITLE
Add registry_deploy role

### PR DESCRIPTION
**Pull request by raukadah**
It will be used to deploy local registry.

This PR has:
- [x] Appropriate testing (molecule, python unit tests)
- [x] Appropriate documentation (README in the role, main README is up-to-date)

### CODE REVIEW

**Comment by raukadah** [comment]
recheck

File: ci_framework/roles/registry_deploy/molecule/default/converge.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+- name: Converge
+  hosts: all
+  pre_tasks:
+    - name: Set uid and gid
+      become: true
```
Comment (cjeanner) [inline ci_framework/roles/registry_deploy/molecule/default/converge.yml:22]: with `become: true`, `sudo` is useless

File: ci_framework/roles/registry_deploy/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+- name: Install Podman package
+  become: true
+  package:
+    name: podman
+    state: present
+
+- name: Deploy the local registry
+  block:
+    - name: 'Deploy the local registry from {{ cifmw_rp_registry_image }}'
+      ansible.builtin.shell: >-
```
Comment (cjeanner) [inline ci_framework/roles/registry_deploy/tasks/main.yml:26]: can't we use the podman module instead?

File: ci_framework/roles/registry_deploy/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        --network=host
+        --name registry
+        -e REGISTRY_HTTP_ADDR={{ cifmw_rp_registry_ip }}:{{ cifmw_rp_registry_port }}
+        -e REGISTRY_LOG_LEVEL=debug
+        {{ cifmw_rp_registry_image }}
+      tags:
+        - skip_ansible_lint
+  rescue:
+    - name: "Deploy the local registry from {{ cifmw_rp_registry_image_fallback }}"
+      ansible.builtin.shell: >-
```
Comment (cjeanner) [inline ci_framework/roles/registry_deploy/tasks/main.yml:38]: same - using the module would probably be better.

File: ci_framework/roles/registry_deploy/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        -e REGISTRY_HTTP_ADDR={{ cifmw_rp_registry_ip }}:{{ cifmw_rp_registry_port }}
+        -e REGISTRY_LOG_LEVEL=debug
+        {{ cifmw_rp_registry_image_fallback }}
+      tags:
+        - skip_ansible_lint
+
+- name: Open port for content provider registry
+  become: true
+  ansible.builtin.command: >-
+        iptables -I INPUT -p tcp --dport {{ cifmw_rp_registry_port }} -j ACCEPT
```
Comment (cjeanner) [inline ci_framework/roles/registry_deploy/tasks/main.yml:52]: `iptables` per se is dead, long live nftables - all of tripleo (and, by extension, podified) is using `nft` CLI.

**Review by cjeanner — CHANGES_REQUESTED** [review]

File: ci_framework/roles/registry_deploy/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        image: "{{ cifmw_rp_registry_image_fallback }}"
+        network: host
+        env:
+          REGISTRY_HTTP_ADDR: "{{ cifmw_rp_registry_ip }}:{{ cifmw_rp_registry_port }}"
+          REGISTRY_LOG_LEVEL: debug
+
+- name: Open port for local registry
+  become: true
+  ansible.builtin.command: >-
+    nft insert rule ip filter INPUT tcp dport {{ cifmw_rp_registry_port }} counter accept
```
Comment (cjeanner) [inline ci_framework/roles/registry_deploy/tasks/main.yml:46]: we can use `inet` instead of `ip` here, in order to cover both IPv4 and IPv6 with one rule.

Comment (cjeanner) [reply ci_framework/roles/registry_deploy/tasks/main.yml:47]: (we'll need some more investigations for the "inet", apparently "filter" doesn't exist therein)

**Review by cjeanner — APPROVED** [review]

File: ci_framework/roles/registry_deploy/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+          REGISTRY_LOG_LEVEL: debug
+
+# TODO(raukadah): Switch to inet in future
+- name: Open port for local registry
+  become: true
+  ansible.builtin.command: >-
+    nft insert rule ip filter INPUT tcp dport {{ cifmw_rp_registry_port }} counter accept
+  changed_when: true
+
+- name: Add the local registry to unqualified-search-registries
```
Comment (cjeanner) [inline ci_framework/roles/registry_deploy/tasks/main.yml:50]: so for the TOML format, it's a bit sad/annoying.... Apparently there was a try to add some support natively in Ansible, but it was denied: https://github.com/ansible/ansible/issues/69120

Fun fact: ansible does support TOML. For the inventory....

If really, we may add the dependency to that "toiletwater" collection in order to get proper TOML support. Lemme know.

**Pull request merged**